utils/generate_types/overrides-test.d.ts

```diff
@@ -207,6 +207,7 @@ export interface TestInfo {
   timeout: number;
   annotations: { type: string, description?: string }[];
   attachments: { name: string, path?: string, body?: Buffer, contentType: string }[];
+  attach(attachment: { name: string, path?: string, body?: Buffer, contentType: string }): Promise<void>;
```

Comment:
Looking at this, perhaps we can simplify the api?
```js
// Infers content type, defaults name to the basename(path)
attach(path: string, options: { contentType?: string, name?: string })
// Defaults to text/plain
attach(name: string, body: string)
// Manual buffer
attach(name: string, body: Buffer, contentType: string)
```

Reply:
Yes! Love it! I iterated a bit while converting some local tests and have arrived at something slightly different since there are `string`s you want to have non `text/plain` content types. (JSON.stringify FTW).

Also made inference uniform across the various versions. `contentType` > inferred from `name` > inferred from `path` (which is applicable to more than just the file path variant).